engines/query-sparql/test/util.ts

```diff
@@ -1,42 +1,26 @@
-/* eslint-disable jest/require-top-level-describe */
-import { resolve } from 'node:path';
-import { Polly } from '@pollyjs/core';
-import { setupPolly } from 'setup-polly-jest';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
 
-const NodeHttpAdapter = require('@pollyjs/adapter-node-http');
-const FSPersister = require('@pollyjs/persister-fs');
+const md5 = require('md5');
 
-const recordingsDir = resolve(__dirname, './assets/http');
+const fetchFn = globalThis.fetch;
 
-Polly.register(FSPersister);
-Polly.register(NodeHttpAdapter);
-
-// Configure everything related to PollyJS
-export function usePolly() {
-  const pollyContext = mockHttp();
-
-  beforeEach(() => {
-    pollyContext.polly.server.any().on('beforePersist', (req, recording) => {
-      recording.request.headers = recording.request.headers.filter(({ name }: any) => name !== 'user-agent');
-    });
-  });
-
-  afterEach(async() => {
-    await pollyContext.polly.flush();
-  });
-}
-
-// Mocks HTTP requests using Polly.JS
-export function mockHttp() {
-  return setupPolly({
-    adapters: [ NodeHttpAdapter ],
-    persister: FSPersister,
-    persisterOptions: { fs: { recordingsDir }},
-    recordFailedRequests: true,
-    matchRequestsBy: {
-      headers: {
-        exclude: [ 'user-agent' ],
-      },
-    },
-  });
-}
+export async function fetch(...args: Parameters<typeof fetchFn>): ReturnType<typeof fetchFn> {
+  // eslint-disable-next-line ts/no-base-to-string
+  const pth = path.join(__dirname, 'networkCache', md5(args[0].toString()));
+  if (!fs.existsSync(pth)) {
+    const res = await fetchFn(...args);
+    const headersObject: Record<string, string> = {};
+    for (const key in res.headers) {
+      headersObject[key] = res.headers.get(key)!;
+    }
+    fs.writeFileSync(pth, JSON.stringify({
+      content: await res.text().catch(() => ''),
```

Comment:
We need to make sure we forward the error here, and probably don't cache the file then.